### DISCUSSION TITLE
Change apollo client by computed property

### DISCUSSION THREAD

**Discussion by sashberd** [Akryum/vue-apollo#1292] [Q&A]
Hi,
I faced with issue in multiple apollo client app when I need to change apollo client dynamically according computed/vuex getter value. (In my case some boolean calculation)
However, it cannot be used, since, as I understand, computed property is not ready when apollo executes the query, thus I always run with "default client"

here is code below:
```
export default {
 computed: {
 isUserAuthenticated() {
      return (
        this.isAuthenticated && this.user.isAdmin
      );
    },
 ...mapGetters({   isAuthenticated: 'isAuthenticated',})
}

 apollo: {
    getCustomAdminData: {
      query: userRowId,
      client: (this.isUserAuthenticated && 'apolloAdminClient') || 'defaultClient',
      deep: false,
      variables() {
        return {
          userID: this.userId
        };
      },
      update(data) {
        return   ...
      },
    },
}
```

I found some remark in this [issue](https://github.com/vuejs/vue-apollo/issues/216). However, the provided solution as

```
 $client() {
        return (this.isUserAuthenticated && 'apolloAdminClient') || 'defaultClient';
      },
```
Is there any solution or workaround in order to get apollo client be dynamic?
Thanks

"vue-apollo": "^3.0.8",

**Comment by Akryum** [comment]
> However, it cannot be used, since, as I understand, computed property is not ready when apollo executes the query, thus I always run with "default client"

The apollo queries are set up in the `created` component lifecycle hook, which means that all the reactivity (including computed properties) are already set up and available. Your issue might be somewhere else.

![image](https://user-images.githubusercontent.com/2798204/143778795-ac186258-91ff-4f69-95f6-5c7c721aab07.png)

**Comment by Akryum** [comment]
There are several issues in your code sample:

```js
client: (this.isUserAuthenticated && 'apolloAdminClient') || 'defaultClient',
```

This can never be reactive since it's not in a function, and `this` correspond to `window` hence why you always have `'defaultClient'`.

Although you could use a function just like you usually would with `variables`, it won't work as `client` is not part of the reactive variables. You can however wrap the entire query options in a function that will be reactive (and will kill and restart the query with the new options automatically):

```js
export default {
  apollo: {
    getCustomAdminData () {
      return {
        query: userRowId,
        client: this.isUserAuthenticated ? 'apolloAdminClient' : 'defaultClient',
        deep: false,
        variables () {
          return {
            userID: this.userId,
          }
        },
        update (data) {
          // ...
        },
      }
    },
  },
}
```
